### ISSUE TITLE
Many sphinx warnings from demos

### ISSUE THREAD

**Issue by huonw** [stellargraph/stellargraph#1360]
### Describe the bug

Doing a sphinx build emits many warnings, a significant chunk of which are from the demos directory. (See #1154 for the general form that covers the non-demo warnings)

This is related to issues like: #1292, #1293, #1294, #1298 and #1305. Some or all of the warnings may be fixed by addressing those.

### To Reproduce

```shell
cd docs
pip install -r requirements.txt
make html
```

### Observed behavior

Many warnings from the `demos` directory (output filtered to just the ones containing `demos`):

```
.../docs/demos/basics/index.txt:4: WARNING: toctree glob pattern '*/index' didn't match any documents
.../docs/demos/basics/loading-networkx.nblink:1423: WARNING: File not found: 'demos/README.md'
.../docs/demos/basics/loading-pandas.nblink:4554: WARNING: File not found: 'demos/README.md'
.../docs/demos/basics/loading-saving-neo4j.nblink:2598: WARNING: File not found: 'demos/README.md'
.../docs/demos/basics/loading-saving-neo4j.nblink:2598: WARNING: File not found: 'demos/connector/neo4j/readme.md'
.../docs/demos/calibration/index.txt:4: WARNING: toctree glob pattern '*/index' didn't match any documents
.../docs/demos/community_detection/index.txt:4: WARNING: toctree glob pattern '*/index' didn't match any documents
.../docs/demos/connector/index.txt:4: WARNING: toctree glob pattern './*' didn't match any documents
.../docs/demos/connector/neo4j/index.txt:4: WARNING: toctree glob pattern '*/index' didn't match any documents
.../docs/demos/embeddings/deep-graph-infomax-cora.nblink:548: WARNING: Title level inconsistent:
.../docs/demos/embeddings/deep-graph-infomax-cora.nblink:597: WARNING: Title level inconsistent:
.../docs/demos/embeddings/index.txt:4: WARNING: toctree glob pattern '*/index' didn't match any documents
.../docs/demos/ensembles/index.txt:4: WARNING: toctree glob pattern '*/index' didn't match any documents
.../docs/demos/graph-classification/index.txt:4: WARNING: toctree glob pattern '*/index' didn't match any documents
.../docs/demos/graph-classification/supervised-graph-classification.nblink:219: WARNING: image file not readable: demos/graph-classification/graph_classification_architecture.png
.../docs/demos/index.txt:4: WARNING: toctree glob pattern './*' didn't match any documents
.../docs/demos/interpretability/gat/index.txt:4: WARNING: toctree glob pattern '*/index' didn't match any documents
.../docs/demos/interpretability/gcn/hateful-twitters-interpretability.nblink:1303: WARNING: Title level inconsistent:
.../docs/demos/interpretability/gcn/hateful-twitters-interpretability.nblink:169: WARNING: Unknown target name: "c".
.../docs/demos/interpretability/gcn/hateful-twitters-interpretability.nblink:171: WARNING: Unknown target name: "c".
.../docs/demos/interpretability/gcn/hateful-twitters-interpretability.nblink:173: WARNING: Unknown target name: "c".
.../docs/demos/interpretability/gcn/hateful-twitters-interpretability.nblink:175: WARNING: Unknown target name: "c".
.../docs/demos/interpretability/gcn/hateful-twitters-interpretability.nblink:177: WARNING: Unknown target name: "c".
.../docs/demos/interpretability/gcn/hateful-twitters-interpretability.nblink:180: WARNING: Unknown target name: "c".
.../docs/demos/interpretability/gcn/hateful-twitters-interpretability.nblink:182: WARNING: Unknown target name: "c".
.../docs/demos/interpretability/gcn/hateful-twitters-interpretability.nblink:184: WARNING: Unknown target name: "c".
.../docs/demos/interpretability/gcn/hateful-twitters-interpretability.nblink:186: WARNING: Unknown target name: "c".
.../docs/demos/interpretability/gcn/hateful-twitters-interpretability.nblink:188: WARNING: Unknown target name: "c".
.../docs/demos/interpretability/gcn/index.txt:4: WARNING: toctree glob pattern '*/index' didn't match any documents
.../docs/demos/interpretability/index.txt:4: WARNING: toctree glob pattern './*' didn't match any documents
.../docs/demos/link-prediction/attri2vec/index.txt:4: WARNING: toctree glob pattern '*/index' didn't match any documents
.../docs/demos/link-prediction/gcn/index.txt:4: WARNING: toctree glob pattern '*/index' didn't match any documents
.../docs/demos/link-prediction/graphsage/index.txt:4: WARNING: toctree glob pattern '*/index' didn't match any documents
.../docs/demos/link-prediction/hinsage/index.txt:4: WARNING: toctree glob pattern '*/index' didn't match any documents
.../docs/demos/link-prediction/index.txt:4: WARNING: toctree glob pattern './*' didn't match any documents
.../docs/demos/link-prediction/knowledge-graphs/index.txt:4: WARNING: toctree glob pattern '*/index' didn't match any documents
.../docs/demos/link-prediction/random-walks/index.txt:4: WARNING: toctree glob pattern '*/index' didn't match any documents
.../docs/demos/node-classification/attri2vec/index.txt:4: WARNING: toctree glob pattern '*/index' didn't match any documents
.../docs/demos/node-classification/cluster-gcn/cluster-gcn-node-classification.nblink:479: WARNING: Title level inconsistent:
.../docs/demos/node-classification/cluster-gcn/index.txt:4: WARNING: toctree glob pattern '*/index' didn't match any documents
.../docs/demos/node-classification/gat/index.txt:4: WARNING: toctree glob pattern '*/index' didn't match any documents
.../docs/demos/node-classification/gcn/gcn-cora-node-classification-example.nblink:80: WARNING: File not found: 'demos/node-classification/README.md'
.../docs/demos/node-classification/gcn/gcn-cora-node-classification-example.nblink:80: WARNING: File not found: 'demos/README.md'
.../docs/demos/node-classification/gcn/gcn-cora-node-classification-example.nblink:80: WARNING: File not found: 'demos/link-prediction/README.md'
.../docs/demos/node-classification/gcn/gcn-cora-node-classification-example.nblink:80: WARNING: File not found: 'demos/community_detection/README.md'
.../docs/demos/node-classification/gcn/gcn-cora-node-classification-example.nblink:80: WARNING: File not found: 'demos/embeddings/README.md'
.../docs/demos/node-classification/gcn/gcn-cora-node-classification-example.nblink:560: WARNING: File not found: 'demos/README.md'
.../docs/demos/node-classification/gcn/gcn-cora-node-classification-example.nblink:1703: WARNING: File not found: 'demos/node-classification/README.md'
.../docs/demos/node-classification/gcn/gcn-cora-node-classification-example.nblink:1703: WARNING: File not found: 'demos/README.md'
.../docs/demos/node-classification/gcn/index.txt:4: WARNING: toctree glob pattern '*/index' didn't match any documents
.../docs/demos/node-classification/graphsage/index.txt:4: WARNING: toctree glob pattern '*/index' didn't match any documents
.../docs/demos/node-classification/index.txt:4: WARNING: toctree glob pattern './*' didn't match any documents
.../docs/demos/node-classification/node2vec/index.txt:4: WARNING: toctree glob pattern '*/index' didn't match any documents
.../docs/demos/node-classification/ppnp/index.txt:4: WARNING: toctree glob pattern '*/index' didn't match any documents
.../docs/demos/node-classification/rgcn/index.txt:4: WARNING: toctree glob pattern '*/index' didn't match any documents
.../docs/demos/node-classification/sgc/index.txt:4: WARNING: toctree glob pattern '*/index' didn't match any documents
.../docs/demos/use-cases/hateful-twitters.nblink:2236: WARNING: Title level inconsistent:
.../docs/demos/use-cases/hateful-twitters.nblink:2279: WARNING: Title level inconsistent:
.../docs/demos/use-cases/hateful-twitters.nblink:2527: WARNING: Title level inconsistent:
.../docs/demos/use-cases/hateful-twitters.nblink:2621: WARNING: Title level inconsistent:
.../docs/demos/use-cases/hateful-twitters.nblink:3020: WARNING: Title level inconsistent:
.../docs/demos/use-cases/hateful-twitters.nblink:3055: WARNING: Title level inconsistent:
.../docs/demos/use-cases/hateful-twitters.nblink:3105: WARNING: Title level inconsistent:
.../docs/demos/use-cases/hateful-twitters.nblink:3199: WARNING: Title level inconsistent:
.../docs/demos/use-cases/hateful-twitters.nblink:3281: WARNING: Title level inconsistent:
.../docs/demos/use-cases/hateful-twitters.nblink:186: WARNING: Unknown target name: "c".
.../docs/demos/use-cases/hateful-twitters.nblink:188: WARNING: Unknown target name: "c".
.../docs/demos/use-cases/hateful-twitters.nblink:190: WARNING: Unknown target name: "c".
.../docs/demos/use-cases/hateful-twitters.nblink:192: WARNING: Unknown target name: "c".
.../docs/demos/use-cases/hateful-twitters.nblink:194: WARNING: Unknown target name: "c".
.../docs/demos/use-cases/hateful-twitters.nblink:197: WARNING: Unknown target name: "c".
.../docs/demos/use-cases/hateful-twitters.nblink:199: WARNING: Unknown target name: "c".
.../docs/demos/use-cases/hateful-twitters.nblink:201: WARNING: Unknown target name: "c".
.../docs/demos/use-cases/hateful-twitters.nblink:203: WARNING: Unknown target name: "c".
.../docs/demos/use-cases/hateful-twitters.nblink:205: WARNING: Unknown target name: "c".
.../docs/demos/use-cases/index.txt:4: WARNING: toctree glob pattern '*/index' didn't match any documents
```

### Expected behavior

No warnings.

### Environment

Operating system: `Darwin-18.6.0-x86_64-i386-64bit`
Python version:
```
3.6.9 (default, Jul 10 2019, 12:25:55) 
[GCC 4.2.1 Compatible Apple LLVM 10.0.1 (clang-1001.0.46.4)]
```
Package versions:
<details>

```
absl-py==0.9.0
alabaster==0.7.12
appdirs==1.4.3
appnope==0.1.0
astor==0.8.1
attrs==19.3.0
Babel==2.8.0
backcall==0.1.0
black==19.10b0
bleach==3.1.0
boto==2.49.0
boto3==1.10.45
botocore==1.13.45
cachetools==4.0.0
certifi==2019.11.28
chardet==3.0.4
Click==7.0
cloudpickle==1.2.2
commonmark==0.9.1
coverage==4.5.4
coveralls==1.10.0
cycler==0.10.0
decorator==4.4.1
defusedxml==0.6.0
docopt==0.6.2
docutils==0.15.2
doit==0.31.1
downhill==0.4.0
entrypoints==0.3
gast==0.2.2
gensim==3.8.1
google-auth==1.10.0
google-auth-oauthlib==0.4.1
google-pasta==0.1.8
grpcio==1.26.0
h5py==2.10.0
idna==2.8
imagesize==1.2.0
import-deps==0.1.0
importlib-metadata==1.3.0
ipykernel==5.1.3
ipython==7.11.1
ipython-genutils==0.2.0
ipywidgets==7.5.1
isodate==0.6.0
jedi==0.15.2
Jinja2==2.10.3
jmespath==0.9.4
joblib==0.14.1
jsonschema==3.2.0
jupyter==1.0.0
jupyter-client==5.3.4
jupyter-console==6.0.0
jupyter-core==4.6.1
Keras-Applications==1.0.8
Keras-Preprocessing==1.1.0
keyring==21.1.0
kiwisolver==1.1.0
llvmlite==0.30.0
MacFSEvents==0.8.1
Markdown==3.0.1
MarkupSafe==1.1.1
matplotlib==3.1.2
mistune==0.8.4
more-itertools==8.0.2
mplleaflet==0.0.5
nbconvert==5.6.1
nbformat==4.4.0
nbsphinx==0.6.1
nbsphinx-link==1.3.0
networkx==2.4
notebook==6.0.2
numba==0.46.0
numpy==1.18.0
oauthlib==3.1.0
opt-einsum==3.1.0
packaging==19.2
pandas==0.25.3
pandocfilters==1.4.2
parso==0.5.2
pathspec==0.7.0
pexpect==4.7.0
pickleshare==0.7.5
pkginfo==1.5.0.1
pluggy==0.13.1
prometheus-client==0.7.1
prompt-toolkit==2.0.10
protobuf==3.11.2
ptyprocess==0.6.0
py==1.8.1
py-cpuinfo==5.0.0
pyasn1==0.4.8
pyasn1-modules==0.2.7
Pygments==2.5.2
Pympler==0.8
pyparsing==2.4.6
pyrsistent==0.15.6
pytest==5.3.1
pytest-benchmark==3.2.2
pytest-cov==2.8.1
pytest-incremental==0.5.0
python-dateutil==2.8.1
python-igraph==0.7.1.post6
pytz==2019.3
pyzmq==18.1.1
qtconsole==4.6.0
rdflib==4.2.2
readme-renderer==24.0
recommonmark==0.6.0
regex==2020.1.8
requests==2.22.0
requests-oauthlib==1.3.0
requests-toolbelt==0.9.1
rsa==4.0
s3transfer==0.2.1
scikit-learn==0.22
scipy==1.4.1
seaborn==0.9.0
Send2Trash==1.5.0
six==1.13.0
smart-open==1.9.0
snowballstemmer==2.0.0
Sphinx==2.3.1
sphinx-markdown-tables==0.0.12
sphinx-rtd-theme==0.4.3
sphinxcontrib-applehelp==1.0.1
sphinxcontrib-devhelp==1.0.1
sphinxcontrib-htmlhelp==1.0.2
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.2
sphinxcontrib-serializinghtml==1.1.3
-e git+git@github.com:stellargraph/stellargraph.git@b6632721e92a590ad511bb8b66e7f5d0d9b043fa#egg=stellargraph&subdirectory=../../../stellargraph-4
tensorboard==2.1.1
tensorflow==2.1.0
tensorflow-estimator==2.1.0
termcolor==1.1.0
terminado==0.8.3
testpath==0.4.4
Theano==1.0.4
toml==0.10.0
tornado==6.0.3
tqdm==4.41.1
traitlets==4.3.3
treon==0.1.3
twine==3.1.1
typed-ast==1.4.1
urllib3==1.25.7
wcwidth==0.1.8
webencodings==0.5.1
Werkzeug==0.16.0
widgetsnbextension==3.5.1
wrapt==1.11.2
zipp==0.6.0

```

</details>

### Additional context

N/A

**Comment by timpitman** [comment]
See also #1432 which could make warnings more visible

**Comment by huonw** [comment]
Looking at https://readthedocs.org/projects/stellargraph/builds/10961639/, there's a few warnings that would be good to resolve for 1.0:

```
/home/docs/checkouts/readthedocs.org/user_builds/stellargraph/checkouts/1456/docs/demos/ensembles/ensemble-link-prediction-example.nblink:1465: WARNING: Error in "image" directive:
no content permitted.
/home/docs/checkouts/readthedocs.org/user_builds/stellargraph/checkouts/latest/docs/demos/graph-classification/dgcnn-graph-classification.nblink:162: WARNING: image file not readable: demos/graph-classification/dgcnn_architecture.png
/home/docs/checkouts/readthedocs.org/user_builds/stellargraph/checkouts/1456/docs/demos/graph-classification/supervised-graph-classification.nblink:163: WARNING: image file not readable: demos/graph-classification/graph_classification_architecture.png
```

There's some other ones that will be resolved by removing subfolders (#1428).
